pkg/intoto/verify.go

```diff
@@ -0,0 +1,53 @@
+package intoto
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/cnabio/signy/pkg/docker"
+	log "github.com/sirupsen/logrus"
+	"github.com/theupdateframework/notary/client"
+)
+
+const (
+	// Fixed as per CNAB-100 spec
+	BundleFilename = "bundle.json"
+	ReadOnlyMask   = 0400
+)
+
+func Verify(verificationImage string, target *client.TargetWithRole, bundle []byte, logLevel string) error {
+	m := &Metadata{}
+	err := json.Unmarshal(*target.Custom, m)
+	if err != nil {
+		return err
+	}
+
+	verificationDir, err := ioutil.TempDir(os.TempDir(), "in-toto")
+	if err != nil {
+		return err
+	}
+	defer func() {
+		os.RemoveAll(verificationDir)
+		os.Remove(verificationDir)
+	}()
+
+	log.Infof("Writing in-toto metadata files into %v", verificationDir)
+	err = WriteMetadataFiles(m, verificationDir)
+	if err != nil {
+		return err
+	}
+
+	bundleFilename := filepath.Join(verificationDir, BundleFilename)
+	err = ioutil.WriteFile(bundleFilename, bundle, ReadOnlyMask)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if verificationImage == "" {
```

Comment:
(We should definitely discuss this in the next CNAB meeting - already on the agenda)
I have some reservations around UX here - i.e. not passing an image vs. actively passing a flag that requests the verification to be performed locally.

On top of that, I think we could look into a Verifier interface that has implementations locally and inside a container, and choose the implementation based on the flag.

Reply:
Fixed! Now there's a boolean flag to control whether verification happens on the OS or in the container. Please take a look?